pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -59,9 +59,8 @@ func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.H
 type InsecureServingInfo struct {
 	// BindAddress is the ip:port to serve on
 	BindAddress string
-	// BindNetwork is the type of network to bind to - defaults to "tcp", accepts "tcp",
-	// "tcp4", and "tcp6".
-	BindNetwork string
+	// Listener is the insecure server network listener.
+	Listener net.Listener
```

Comment:
I wonder whether we want both ways: either `Bind*` (including BindNetwork) are set, or `Listener` is set. If the former case we create the listener in ApplyTo, in the later we just use it, but panic if `Bind*` and `Listener` are set.

Reply:
Same for secure serving as well of course.